File: web/web-frontend/src/app/generic/error/report-modal/model/error.feedback.ts
import {Serializable} from "../../../../model/infrastructure/serializable.model";
import {JsonUtil} from "../../../../utils/json.util";

export class ErrorFeedback implements Serializable<ErrorFeedback> {

    contactName: string;
    contactEmail: string;
    stepsToReproduce: string;

    errorMessage: string;
    errorStacktrace: string;

    deserialize(input: Object): ErrorFeedback {
        this.contactName = input['contactName'];
        this.contactEmail = input['contactEmail'];
        this.stepsToReproduce = input['stepsToReproduce'];
        this.errorMessage = input['errorMessage'];
        this.errorStacktrace = input['errorStacktrace'];
        return this;
    }

    serialize(): string {
        return "" +
            '{' +
            '"contactName":' + JsonUtil.stringify(this.contactName) +
            ',"contactEmail":' + JsonUtil.stringify(this.contactEmail) +
            ',"stepsToReproduce":' + JsonUtil.stringify(this.stepsToReproduce) +
            ',"errorMessage":' + JsonUtil.stringify(this.errorMessage) +
            ',"errorStacktrace":' + JsonUtil.stringify(this.errorStacktrace) +
            '}'
    }
}

File: model/src/main/kotlin/com/testerum/model/feedback/ErrorFeedback.kt
package com.testerum.model.feedback

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty

data class ErrorFeedback @JsonCreator constructor(@JsonProperty("contactName") val contactName: String?,
                                                  @JsonProperty("contactEmail") val contactEmail: String?,
                                                  @JsonProperty("stepsToReproduce") val stepsToReproduce: String?,
                                                  @JsonProperty("errorMessage") val errorMessage: String?,
                                                  @JsonProperty("errorStacktrace") val errorStacktrace: String?)
